Mathlib/NumberTheory/NumberField/FinitePlaces.lean

```diff
@@ -0,0 +1,242 @@
+  norm_cast
+  by_contra! h
+  apply IsPrime.ne_top v.isPrime
+  rw [← absNorm_eq_one_iff]
+  have : 0 < absNorm v.asIdeal := by
+    rw [Nat.pos_iff_ne_zero, absNorm_ne_zero_iff]
+    have : Fintype (𝓞 K ⧸ v.asIdeal) := Ideal.fintypeQuotientOfFreeOfNeBot v.asIdeal v.ne_bot
+    exact Fintype.finite this
+  omega
+
+private lemma norm_ne_zero : (absNorm v.asIdeal : NNReal) ≠ 0 := ne_zero_of_lt (one_lt_norm v)
+
+/-- The `v`-adic absolute value on `K` defined as the norm of `v` raised to negative `v`-adic
+valuation.-/
+noncomputable def vadic_abv : AbsoluteValue K ℝ where
+  toFun := fun x ↦ toNNReal (norm_ne_zero v) (v.valuation x)
+  map_mul' := fun _ _ ↦ by simp only [_root_.map_mul, NNReal.coe_mul]
+  nonneg' := fun _ ↦ NNReal.zero_le_coe
+  eq_zero' := fun _ ↦ by simp only [NNReal.coe_eq_zero, map_eq_zero]
+  add_le' := by
+    intro x y
+    simp only
+    norm_cast
+    -- the triangular inequality is implied by the ultrametric one
+    apply le_trans _ <| max_le_add_of_nonneg (zero_le ((toNNReal (norm_ne_zero v)) (v.valuation x)))
+      (zero_le ((toNNReal (norm_ne_zero v)) (v.valuation y)))
+    have h_mono := StrictMono.monotone (toNNReal_strictMono (one_lt_norm v))
+    rw [← Monotone.map_max h_mono] --max goes inside withZeroMultIntToNNReal
+    exact h_mono (Valuation.map_add v.valuation x y)
+
+theorem vadic_abv_def {x : K} : vadic_abv v x = toNNReal (norm_ne_zero v) (v.valuation x) := rfl
+
+end absoluteValue
+
+section FinitePlace
+variable {K : Type*} [Field K] [NumberField K] (v : HeightOneSpectrum (𝓞 K))
+
+/-- The embedding of a NumberField inside its completion with respect to `v`. -/
+def embedding : K →+* adicCompletion K v :=
+    @UniformSpace.Completion.coeRingHom K _ v.adicValued.toUniformSpace _ _
+
+noncomputable instance instRankOneValuedAdicCompletion :
+    Valuation.RankOne (valuedAdicCompletion K v).v where
+  hom := {
+    toFun := toNNReal (norm_ne_zero v)
+    map_zero' := rfl
+    map_one' := rfl
+    map_mul' := MonoidWithZeroHom.map_mul (toNNReal (norm_ne_zero v))
+  }
+  strictMono' := toNNReal_strictMono (one_lt_norm v)
+  nontrivial' := by
+    rcases Submodule.exists_mem_ne_zero_of_ne_bot v.ne_bot with ⟨x, hx1, hx2⟩
+    use (x : K)
+    rw [valuedAdicCompletion_eq_valuation' v (x : K)]
+    constructor
+    · simpa only [ne_eq, map_eq_zero, NoZeroSMulDivisors.algebraMap_eq_zero_iff]
+    · apply ne_of_lt
+      rw [valuation_eq_intValuationDef, intValuation_lt_one_iff_dvd]
+      exact dvd_span_singleton.mpr hx1
+
+/-- The `v`-adic completion of `K` is a normed field. -/
+noncomputable instance instNormedFieldValuedAdicCompletion : NormedField (adicCompletion K v) :=
+    Valued.toNormedField (adicCompletion K v) (WithZero (Multiplicative ℤ))
+
+/-- A finite place of a number field `K` is a place associated to an embedding into a completion
+with respect to a maximal ideal. -/
+def FinitePlace (K : Type*) [Field K] [NumberField K] := {w : AbsoluteValue K ℝ //
```

Comment:
```suggestion
def FinitePlace (K : Type*) [Field K] [NumberField K] :=
  {w : AbsoluteValue K ℝ //
```
If the definition doesn't fit on the remainder of the line, it should start in the following line (with 2 spaces indentation).